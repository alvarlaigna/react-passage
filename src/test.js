import React from 'react'
import { MemoryRouter, Switch, Route } from 'react-router-dom'
import { Passage, Link, Redirect } from './index'
import renderer from 'react-test-renderer'

describe('react-passage', () => {
  it('renders an anchor tag without context', () => {
    const component = renderer.create(<Link to='/blades'>Blades</Link>)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders an anchor tag when a route is unmatched', () => {
    const component = renderer.create(
      <Passage>
        <MemoryRouter>
          <Switch>
            <Route path='/get-started' exact render={() => {}} />
            <Route render={() => <Link to='/blades'>Blades</Link>} />
          </Switch>
        </MemoryRouter>
      </Passage>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders a Link tag when a route is matched', () => {
    const component = renderer.create(
      <Passage>
        <MemoryRouter>
          <Switch>
            <Route path='/get-started/plan/:planId' exact render={() => {}} />
            <Route path='/users/:id' exact render={() => {}} />

            <Route render={() => (
              <div>
                <Link to='/get-started/plan/shave'>Shave Core</Link>
                <Link to='/users/6'>User 6</Link>
              </div>
            )} />
          </Switch>
        </MemoryRouter>
      </Passage>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('redirects to a route without context', () => {
    const mockVia = jest.fn()
    renderer.create(<Redirect to='/our-products/shave' via={mockVia} />)
    expect(mockVia).toHaveBeenCalledWith('/our-products/shave')
  })

  it('redirects to a matched react route', () => {
    const component = renderer.create(
      <Passage>
        <MemoryRouter>
          <Switch>
            <Route path='/get-started' exact render={() => <p>Redirected!</p>} />
            <Route render={() => <Redirect to='/get-started' />} />
          </Switch>
        </MemoryRouter>
      </Passage>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('redirects to an unmatched react route', () => {
    const mockVia = jest.fn()
    renderer.create(
      <Passage>
        <MemoryRouter>
          <Switch>
            <Route path='/get-started' exact render={() => <p>Redirected!</p>} />
            <Route render={() => <Redirect to='/our-products' via={mockVia} />} />
          </Switch>
        </MemoryRouter>
      </Passage>
    )
    expect(mockVia).toHaveBeenCalledWith('/our-products')
  })
})
