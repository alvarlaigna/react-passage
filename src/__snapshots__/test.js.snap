// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-passage redirects to a matched react route 1`] = `
<p>
  Redirected!
</p>
`;

exports[`react-passage renders a Link tag when a route is matched 1`] = `
<div>
  <a
    data-safelink-type="link"
    href="/get-started/plan/shave"
    onClick={[Function]}
  >
    Shave Core
  </a>
  <a
    data-safelink-type="link"
    href="/users/6"
    onClick={[Function]}
  >
    User 6
  </a>
</div>
`;

exports[`react-passage renders an anchor tag when a route is unmatched 1`] = `
<a
  data-safelink-type="a"
  href="/blades"
>
  Blades
</a>
`;

exports[`react-passage renders an anchor tag without context 1`] = `
<a
  data-safelink-type="a"
  href="/blades"
>
  Blades
</a>
`;
